lib/backend/k8s/resources/resources.go

```diff
@@ -100,3 +102,74 @@ func SetCalicoMetadataFromK8sAnnotations(calicoRes Resource, k8sRes Resource) {
 		}
 	}
 }
+
+// Store Calico Metadata in the in the k8s resource annotations for CRD backed resources.
+// This should store all Metadata except for those stored in Annotations and Labels and
+// store them in annotations.
+func StoreCalicoMetadataInK8sAnnotations(res Resource) error {
+	metadataBytes, err := json.Marshal(res.GetObjectMeta().(*metav1.ObjectMeta))
+	if err != nil {
+		return err
+	}
+	annotations := res.GetObjectMeta().GetAnnotations()
+	if len(annotations) == 0 {
+		annotations = make(map[string]string)
+	}
+	annotations[metadataAnnotation] = string(metadataBytes)
+	res.GetObjectMeta().SetAnnotations(annotations)
+
+	// Make sure to clear out all of the Calico Metadata out of the ObjectMeta except for
+	// Name, Namespace, and ResourceVersion.
+	res.GetObjectMeta().SetSelfLink("")
```

Comment:
this would have been less code if you'd first got the ObjectMeta in a local variable (and potentially less operations depending on how clever the compiler is)